tests/tests.py

```diff
@@ -938,3 +952,425 @@ def test_invalidate_model_with_geometry(self):
+                        self.assertTrue(in_transaction())
+                        with self.assertNumQueries(1):
+                            self.assertEqual(
+                                'Django',
+                                list(Category.objects.filter(pk=1).cache())[0].title
+                            )
+                        self.assertEqual(
+                            'Django',
+                            return_from_other_thread(
+                                lambda: list(Category.objects.filter(pk=1).cache())[0].title
+                            )
+                        )
+                        obj = list(Category.objects.filter(pk=1).cache())[0]
+                        obj.title = 'Changed'
+                        obj.save()
+                        invalidate_all()
+                        with self.assertNumQueries(1):
+                            self.assertEqual(
+                                'Changed',
+                                list(Category.objects.filter(pk=1).cache())[0].title
+                            )
+                        self.assertEqual(
+                            'Django',
+                            return_from_other_thread(
+                                lambda: list(Category.objects.filter(pk=1).cache())[0].title
+                            )
+                        )
+                        queue_item = list(_function_queue)[-1]
+                        self.assertEqual('invalidate_all', queue_item['func'].__name__)
+                        raise IntentionalRollback()
+                except IntentionalRollback:
+                    pass
+                with self.assertNumQueries(1):
+                    self.assertEqual('Django', list(Category.objects.filter(pk=1).cache())[0].title)
+                self.assertEqual(
+                    'Django',
+                    return_from_other_thread(
+                        lambda: list(Category.objects.filter(pk=1).cache())[0].title
+                    )
+                )
+                self.assertEqual(0, len(list(_function_queue)))
+                self.assertTrue(in_transaction())
+            with self.assertNumQueries(1):
+                self.assertEqual('Django', list(Category.objects.filter(pk=1).cache())[0].title)
+            self.assertEqual(
+                'Django',
+                return_from_other_thread(
+                    lambda: list(Category.objects.filter(pk=1).cache())[0].title
+                )
+            )
+            self.assertEqual(0, len(list(_function_queue)))
+            self.assertTrue(in_transaction())
+        self.assertFalse(in_transaction())
+        with self.assertNumQueries(0):
+            self.assertEqual('Django', list(Category.objects.filter(pk=1).cache())[0].title)
+        self.assertEqual(
+            'Django',
+            return_from_other_thread(
+                lambda: list(Category.objects.filter(pk=1).cache())[0].title
+            )
+        )
```

Comment:
Please move all transactions tests to separate file like [here](https://github.com/Suor/django-cacheops/compare/transactions-tests).